spec/index/multi_index_spec.rb

```diff
@@ -35,6 +35,30 @@
           labels: [[0,0,1,1,2,2]])
       }.to raise_error
     end
+
+    context "create an MultiIndex with name" do
+      context "if no name is set" do
+        let(:mi) { Daru::MultiIndex.new(
+                    levels: [[:a,:b,:c], [:one, :two]],
+                    labels: [[0,0,1,1,2,2], [0,1,0,1,0,1]]) }
+        it { expect(mi.name).to be_nil }
+      end
+
+      context "correctly return the MultiIndex name" do
+        let(:mi) { Daru::MultiIndex.new(
+                  levels: [[:a,:b,:c], [:one, :two]],
+                  labels: [[0,0,1,1,2,2], [0,1,0,1,0,1]], name: ['n1', 'n2']) }
+        it { expect(mi.name).to eq ['n1', 'n2'] }
```

Comment:
> @zverok , is it okay?

No, it is not :)
1. Please add the tests for both cases (lesser names/more names than expected).
2. Please, make the error reradable (something like "too many names for an index" or something) _and_ raised on name assigning!
3. Please, document the behavior (e.g. something like "if number of names is less than number of index levels, it would be considered ... something and something")